test/index.js

```diff
@@ -117,6 +119,59 @@ describe('Bell', () => {
         });
     });
 
+    it('authenticates an endpoint via oauth using RSA-SHA1 signing', (done) => {
+
+        const mock = new Mock.V1();
+        mock.start((provider) => {
+
+            const server = new Hapi.Server();
+            server.connection({ host: 'localhost', port: 80 });
+            server.register(Bell, (err) => {
+
+                expect(err).to.not.exist();
+
+                server.auth.strategy('custom', 'bell', {
+                    password: 'cookie_encryption_password_secure',
+                    isSecure: false,
+                    clientId: 'test',
+                    clientSecret: privateKey,
+                    provider: provider,
+                    signatureMethod: 'RSA-SHA1'
+                });
+
+                server.route({
+                    method: '*',
+                    path: '/login',
+                    config: {
+                        auth: 'custom',
+                        handler: function (request, reply) {
+
+                            reply(request.auth.credentials);
+                        }
+                    }
+                });
+
+                server.inject('/login?next=%2Fhome', (res) => {
+
+                    const cookie = res.headers['set-cookie'][0].split(';')[0] + ';';
+                    expect(res.headers.location).to.equal(mock.uri + '/auth?oauth_token=1');
+
+                    mock.server.inject(res.headers.location, (mockRes) => {
+
+                        expect(mockRes.headers.location).to.equal('http://localhost:80/login?oauth_token=1&oauth_verifier=123');
```

Comment:
Due to the 100% test coverage and the other test cases, I know we exercise the `RSA-SHA1` path here, but I don't remember if there's a way to verify in the test case itself. ie. check the signature is RSA-SHA1? That will be useful long term and make this a slightly better test.